backend/backend/db/database.py

```diff
@@ -1,12 +1,11 @@
 from sqlalchemy import create_engine
 from sqlalchemy.orm import declarative_base
-from sqlalchemy.orm import sessionmaker
+from sqlalchemy.orm import Session
 
 
 # pass empty connection string to use PG* environment variables
 # (see https://www.postgresql.org/docs/current/libpq-envars.html)
 engine = create_engine("postgresql+psycopg2://", future=True)
-
-SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+session = Session(engine)
```

Comment:
A couple of queries here (and with deleting `app/dependencies.py`)

First, if we drop the use `sessionmaker`, do we also want to go back to default `autoflush=True`? I think dropping the other arguments wouldn't make a difference:  `autocommit=False` is already the default, and `bind` can be replaced by passing `engine` as the first parameter.

https://docs.sqlalchemy.org/en/20/orm/session_basics.html#using-a-sessionmaker

Second, as I read the docs, it's not thread-safe to create a single module-level `session` variable, and for FastAPI these should be constructed once per request. FastAPI have a slightly lighter recommendation now, where we can pass in `session: SessionDep` as a parameter instead of `db: Session = Depends(get_db)`:

```python
# in backend.db.database:
def get_session():
    with Session(engine) as session:
        yield session


SessionDep = Annotated[Session, Depends(get_session)]

# in e.g. backend.app.routers.features:
@router.get("/{feature_id}", response_model=schemas.FeatureOut)
def read_feature(feature_id: int, session: SessionDep):
```

https://fastapi.tiangolo.com/tutorial/sql-databases/#create-a-session-dependency

Reply:
Thanks for taking a look. I've updated this to follow the changes in https://github.com/nismod/infra-risk-vis/pull/190, and added `sessionmaker` back in with `autoFlush=True` for all sessions.

Reply:
My mistake, there's a typo in `autoFlush`.